api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -70,6 +77,21 @@ public static Type fromTypeName(String typeString) {
       return TYPES.get(lowerTypeString);
     }
 
+    Matcher geometry = GEOMETRY_PARAMETERS.matcher(typeString);
+    if (geometry.matches()) {
+      String crs = geometry.group(1);
+      return GeometryType.of(crs != null ? crs.trim() : null);
+    }
+
+    Matcher geography = GEOGRAPHY_PARAMETERS.matcher(typeString);
+    if (geography.matches()) {
+      String crs = geography.group(1);
+      String algorithmName = geography.group(2);
+      EdgeAlgorithm algorithm =
+          algorithmName == null ? null : EdgeAlgorithm.fromName(algorithmName.trim());
+      return GeographyType.of(crs != null ? crs.trim() : null, algorithm);
```

Comment:
Are calls to `trim` needed?

Reply:
Removed trim, as the regex was modified to handle trimming.